tinygrad/shape/symbolic.py

```diff
@@ -34,6 +34,13 @@ def iter_idxs(idxs:Tuple[VariableOrNum, ...]) -> Iterator[Tuple[int,...]]:
   def substitute(self, var_vals: Dict[VariableOrNum, Node]) -> Node: raise RuntimeError(self.__class__.__name__)
   def unbind(self) -> Tuple[Node, Optional[int]]: return self.substitute({v: v.unbind()[0] for v in self.vars() if v.val is not None}), None
 
+  # traverse max attribute looking for an int
+  def find_max_int(self) -> int:
+    tryint = self.max
+    while(not isinstance(tryint, int)): tryint = tryint.max
+    assert isinstance(tryint, int), f"tryint must be integer, {tryint=} isn't"
```

Comment:
Can the assertion actually fail? It'd be stuck in the loop if that condition is not met.